[IT-3697] Make sure the correct base image is used

ConsoleCatzirl opened the pull request:
Use the standard base image, not the minimal image or one tuned for specific hardware. Also give `ssm-user` access to run docker.

ConsoleCatzirl committed:
[IT-3697] Make sure the ssm agent is installed

The ssm agent is no longer installed by default on the base AMI,
install it with packer. Also give `ssm-user` access to run docker.

zaro0508 reviewed (changes requested):
> The SSM agent is no longer installed by default on the base AMI

according to the AWS docs, the SSM agent is pre-installed https://docs.aws.amazon.com/systems-manager/latest/userguide/ami-preinstalled-agent.html

could it be that we might not be building on top of the correct AWS base AMI?  or that the service catalog product is not using the correct AMI?

ConsoleCatzirl commented:
@zaro0508 oh, thanks, we _were_ building off of two different amazon base image types, depending on which was most recently updated. The working images were based off of the AWS ECS-optimized image, and the one that didn't have an ssm agent installed was based off of the minimal AWS image.

ConsoleCatzirl committed:
[IT-3697] Make sure we use the standard AMI

The previous regex would match multiple image types, e.g.:
  * al2023-ami-2023.4.20240401.1-kernel-6.1-x86_64
  * al2023-ami-ecs-hvm-2023.0.20240131-kernel-6.1-x86_64
  * al2023-ami-minimal-2023.4.20240416.0-kernel-6.1-x86_64

Ensure we use the standard image so that the ssm agent will be
installed without also installing the ecs agent.

zaro0508 reviewed (changes requested):
the purpose of this PR has completely changed.  can you please update the PR subject and description to reflect what you are doing in this PR?
